verify() method and toString() fix

RyanR712 opened the pull request:
toString() corrections were put in so I had something to make a pull request for so I could merge master into this branch

Commit:
Corrected toString() calls

Commit:
Merge branch 'master' into 52-verify-aeg

Commit:
verify() implementation and toString() changes
verify() checks if child nodes are contained within the caller,
calls itself on the CutNodes contained within,
and checks for any overlapping children.
verify() is now also in AEGTree instead of CutNode.

Also changed the output of toString() methods in
CutNode.ts, AtomNode.ts, Rectangle.ts, Ellipse.ts

Commit:
Merge branch 'master' into 52-verify-aeg

James-Oswald commented on src/AEG/AEGTree.ts:
```diff
@@ -12,13 +12,45 @@ export class AEGTree {
+            }
+
+            //Check for overlaps on the same level
+            for (let j = 0; j < currentCut.children.length; j++) {
+                if (this.overlaps(currentCut.children[i], currentCut.children[j])) {
+                    return false;
+                }
+            }
```
This will check if a child overlaps with itself, returning false every time. additionally, it's extremely inefficient since it checks overlaps of objects twice overlaps is a symmetric relation, i.e., `overlaps(a,b) == overlaps(b,a)` hence you only need to check one of them. You can *probably* fix both of these issues by only checking the upper triangular of the pairing matrix this setup generates, by starting the second loop at `i+1` instead of 0
```suggestion
            for (let j = i+1; j < currentCut.children.length; j++) {
                if (this.overlaps(currentCut.children[i], currentCut.children[j])) {
                    return false;
                }
            }
```

James-Oswald commented on src/AEG/AEGTree.ts:
```diff
@@ -12,13 +12,45 @@ export class AEGTree {
+                return false;
+            }
+
+            //Check one level deeper if the child is a CutNode
+            if (
+                currentCut.children[i] instanceof CutNode &&
+                !this.verifyAEG(currentCut.children[i] as CutNode)
+            ) {
+                return false;
+            }
```
I would make it more obvious this is your recursive case by moving this into its own secondary loop after this loop.

Commit:
Added 9/30 3:10 suggested changes
In AEGTree:
Changed j = 0 to j = i+1 on line 39
Moved the verify() recursive case to its own loop on line 45
